lib/ansible/modules/cloud/vmware/vcenter_license.py

```diff
@@ -177,8 +199,25 @@ def main():
         if key is not None:
             lam = lm.licenseAssignmentManager
             assigned_license = None
+            datacenter = module.params['datacenter']
+            datacenter_obj = None
+            if datacenter:
+                datacenter_obj = pyv.find_datacenter_by_name(datacenter)
+
+            cluster = module.params['cluster_name']
+            if cluster:
+                if datacenter and not datacenter_obj:
```

Comment:
Since `cluster` depends on `datacenter` parameter, I suggest you add a `required_by` in the `AnsibleModule()` call:
```python
required_by=dict(
    datacenter=('cluster',),
),
```

Reply:
User may or may not provide datacenter with cluster in argument spec so I am not sure if this check is required.

Reply:
Yes, indeed, good point.